src/Utils.php

```diff
@@ -921,4 +921,58 @@ public static function is_canary_island( string $state, string $country ): bool
 
 		return false;
 	}
+
+	/**
+	 * Get the frontend locations.
+	 *
+	 * @return array $locations
+	 */
+	public static function get_frontend_locations(): array {
+		$locations = array(
+			'cart_before_checkout'             => array(
+				'woocommerce_proceed_to_checkout',
+				'postnl_before_woocommerce/proceed-to-checkout-block',
+			),
+			'cart_after_checkout'              => array(
+				'woocommerce_after_cart_totals',
+				'postnl_after_woocommerce/proceed-to-checkout-block',
+			),
+			'checkout_before_customer_details' => array(
+				'woocommerce_checkout_before_customer_details',
+			),
+			'checkout_after_customer_details'  => array(
+				'woocommerce_checkout_after_customer_details',
+			),
+			'minicart_before_buttons'          => array(
+				'woocommerce_widget_shopping_cart_before_buttons',
+				'postnl_before_woocommerce/mini-cart-footer-block',
+			),
+			'minicart_after_buttons'           => array(
+				'woocommerce_widget_shopping_cart_after_buttons',
+				'postnl_after_woocommerce/mini-cart-footer-block',
+			),
+		);
+		// Allow filtering of the locations.
+		return apply_filters( 'postnl_frontend_locations', $locations );
+	}
+
+	/**
+	 * Get the frontend location mapping.
+	 *
+	 * @return array $mapping
+	 */
+	public static function get_frontend_location_mapping(): array {
+		$mapping = array(
+			'cart_before_checkout'             => array( 'postnl_cart_auto_render_button', 'postnl_cart_button_placement', 'before_checkout' ),
+			'cart_after_checkout'              => array( 'postnl_cart_auto_render_button', 'postnl_cart_button_placement', 'after_checkout' ),
+			'checkout_before_customer_details' => array( 'postnl_checkout_auto_render_button', 'postnl_checkout_button_placement', 'before_customer_details' ),
+			'checkout_after_customer_details'  => array( 'postnl_checkout_auto_render_button', 'postnl_checkout_button_placement', 'after_customer_details' ),
+			'minicart_before_buttons'          => array( 'postnl_minicart_auto_render_button', 'postnl_minicart_button_placement', 'before_buttons' ),
+			'minicart_after_buttons'           => array( 'postnl_minicart_auto_render_button', 'postnl_minicart_button_placement', 'after_buttons' ),
+		);
+		// Allow filtering of the mapping.
+		return apply_filters( 'postnl_frontend_location_mapping', $mapping );
```

Comment:
```suggestion
		// Allow filtering of the mapping.
		return apply_filters(
		'postnl_frontend_location_mapping',
		array(
			'cart_before_checkout'             => array( 'postnl_cart_auto_render_button', 'postnl_cart_button_placement', 'before_checkout' ),
			'cart_after_checkout'              => array( 'postnl_cart_auto_render_button', 'postnl_cart_button_placement', 'after_checkout' ),
			'checkout_before_customer_details' => array( 'postnl_checkout_auto_render_button', 'postnl_checkout_button_placement', 'before_customer_details' ),
			'checkout_after_customer_details'  => array( 'postnl_checkout_auto_render_button', 'postnl_checkout_button_placement', 'after_customer_details' ),
			'minicart_before_buttons'          => array( 'postnl_minicart_auto_render_button', 'postnl_minicart_button_placement', 'before_buttons' ),
			'minicart_after_buttons'           => array( 'postnl_minicart_auto_render_button', 'postnl_minicart_button_placement', 'after_buttons' ),
		)
		);
```

Since there is no logic, we can shorten it to be like this. What do you think?